### PR TITLE
FAI-639: CF boolean input constraints limitations

### PR DESCRIPTION
**Pull request by kelvah**
JIRA: https://issues.redhat.com/browse/FAI-639

It should not be possible to set up constraints for boolean search inputs.
So we have to differentiate between inputs that can be selected and inputs that support constraints.

Before this PR the constraint button was visible (but disabled) for boolean inputs. Then it disappeared after selecting the row.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

File: ui-packages/packages/trusty/src/components/Templates/Counterfactual/counterfactualReducer.ts [hunk 1]
```diff
@@ -248,6 +248,20 @@ export const isInputTypeSupported = (searchInput: CFSearchInput): boolean => {
...
+  //not allowed for Booleans.
+  if (searchInput.kind === 'UNIT') {
+    switch (typeof searchInput.value) {
+      case 'number':
+        return true;
+    }
+  }
+  return false;
+};
+
```
Comment (kelvah) [inline ui-packages/packages/trusty/src/components/Templates/Counterfactual/counterfactualReducer.ts:264]: I created another utility method to know which input types support constraints (just number for now :no_mouth:)
It's maybe overkill? I didn't know where to put it honestly.

Comment (manstis) [reply ui-packages/packages/trusty/src/components/Templates/Counterfactual/counterfactualReducer.ts:264]: Not an overkill :+1:

File: ui-packages/packages/trusty/src/components/Organisms/CounterfactualTable/CounterfactualTable.tsx [hunk 2]
```diff
@@ -495,11 +502,14 @@ const ConstraintCell = (props: ConstraintCellProps) => {
...
     isInputTypeSupported,
+    isInputConstraintSupported,
     onEditConstraint
   } = props;
 
   const isTypeSupported = isInputTypeSupported(row);
-  const isConstraintEditEnabled = isInputSelectionEnabled && isTypeSupported;
+  const isConstraintSupported = isInputConstraintSupported(row);
+  const isConstraintEditEnabled =
+    isInputSelectionEnabled && isConstraintSupported;
```
Comment (manstis) [inline ui-packages/packages/trusty/src/components/Organisms/CounterfactualTable/CounterfactualTable.tsx:512]: Is `isTypeSupported` no longer needed here?

Comment (kelvah) [reply ui-packages/packages/trusty/src/components/Organisms/CounterfactualTable/CounterfactualTable.tsx:512]: It's still needed to display the _Not yet supported_ message [at 535](https://github.com/kiegroup/kogito-apps/blob/a63da59cf46b5817f6ea2a65509327549c47a9a2/ui-packages/packages/trusty/src/components/Organisms/CounterfactualTable/CounterfactualTable.tsx#L535)

Comment (manstis) [reply ui-packages/packages/trusty/src/components/Organisms/CounterfactualTable/CounterfactualTable.tsx:512]: Yep, I understand that. I meant is it still needed _here_ to set `isConstraintEditEnabled` or does your refactored logic give equivalent function in the UI.

**Review by manstis — APPROVED** [review]
On reflection the logic seems fine 👍

**Review by sjamboro — APPROVED** [review]

**Comment by kelvah** [comment]
Jenkins retest this

**Comment by manstis** [comment]
@kelvah There may be a failing test?
```
[2021-10-19T09:37:53.751Z] [ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.9.1:yarn (yarn test) on project kogito-apps-ui-packages: Failed to run task: 'yarn run test --registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
```

**Comment by kelvah** [comment]
> @kelvah There may be a failing test?
> 
> ```
> [2021-10-19T09:37:53.751Z] [ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.9.1:yarn (yarn test) on project kogito-apps-ui-packages: Failed to run task: 'yarn run test --registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
> ```

You are right, thank you. Updated failing snapshot.

**Pull request merged**